### PR TITLE
Python Program to Check Krishnamurthy Number

### PR DESCRIPTION
**Pull request by Utsav1999**
### **Describe your change:**
Added Python Program to check whether a number is **Krishnamurthy Number or not**.

* [X] Add an algorithm?
* [ ] Fix a bug or typo in an existing algorithm?
* [ ] Documentation change?

### **Checklist:**
* [X] I have read [CONTRIBUTING.md](https://github.com/TheAlgorithms/Python/blob/master/CONTRIBUTING.md).
* [X] This pull request is all my own work -- I have not plagiarized.
* [X] I know that pull requests will not be merged if they fail the automated tests.
* [ ] This PR only changes one algorithm file.  To ease review, please open separate PRs for separate algorithms.
* [X] All new Python files are placed inside an existing directory.
* [X] All filenames are in all lowercase characters with no spaces or dashes.
* [X] All functions and variable names follow Python naming conventions.
* [X] All function parameters and return values are annotated with Python [type hints](https://docs.python.org/3/library/typing.html).
* [ ] All functions have [doctests](https://docs.python.org/3/library/doctest.html) that pass the automated testing.
* [ ] All new algorithms have a URL in its comments that points to Wikipedia or other similar explanation.
* [X] If this pull request resolves one or more open issues then the commit message contains `Fixes: #{$ISSUE_NO}`.

### CODE REVIEW

File: maths/krishnamurthy_number.py [hunk 1]
```diff
@@ -0,0 +1,46 @@
...
+    So, 145 is a Krishnamurthy Number
+"""
+
+
+def factorial(digit: int) -> int:
+    """
+    function to calculate factorial of each digit
+    """
+
+    return 1 if (digit == 0 or digit == 1) else (digit * factorial(digit - 1))
```
Comment (cclauss) [inline maths/krishnamurthy_number.py:18]:
```suggestion
    return 1 if digit in (0, 1) else digit * factorial(digit - 1)
```

File: maths/krishnamurthy_number.py [hunk 2]
```diff
@@ -0,0 +1,46 @@
...
+number itself.
+
+For example: 145 = 1! + 4! + 5!
+    So, 145 is a Krishnamurthy Number
+"""
+
+
+def factorial(digit: int) -> int:
+    """
+    function to calculate factorial of each digit
```
Comment (cclauss) [inline maths/krishnamurthy_number.py:15]: Doctests?  Or use https://docs.python.org/3/library/math.html#math.factorial

File: maths/krishnamurthy_number.py [hunk 3]
```diff
@@ -0,0 +1,46 @@
...
+
+    >>> krishnamurthy(1)
+    True
+    """
+
+    factSum = 0
+    duplicate = number
+    while duplicate > 0:
+        factSum += factorial(duplicate % 10)
+        duplicate //= 10
```
Comment (cclauss) [inline maths/krishnamurthy_number.py:37]: Could be streamlined with https://docs.python.org/3/library/functions.html#divmod

File: maths/perfect_number.py [hunk 4]
```diff
@@ -0,0 +1,34 @@
...
+
+    Start from 1 because dividing by 0 will raise ZeroDivisionError.
+    A number at most can be divisible by the half of the number except the number
+    itself.  For example, 6 is at most can be divisible by 3 except by 6 itself.
+    """
+    return sum(i for i in range(1, ((number // 2) + 1)) if number % i == 0) == number
+
+
+if __name__ == "__main__":
+    print("Program to check whether a number is a Perfect number or not.......")
```
Comment (cclauss) [inline maths/perfect_number.py:32]:
```suggestion
    print("Program to check whether a number is a Perfect number or not...")
```

File: maths/krishnamurthy_number.py [hunk 5]
```diff
@@ -0,0 +1,46 @@
...
+    return factSum == number
+
+
+if __name__ == "__main__":
+    print("Program to check whether a number is a Krisnamurthy Number or not")
+    number = int(input("Enter number: ").strip())
+    print(
+        f"{number} is {'' if krishnamurthy(number) else 'not '} a Krishnamurthy Number."
```
Comment (cclauss) [inline maths/krishnamurthy_number.py:45]:
```suggestion
        f"{number} is {'' if krishnamurthy(number) else 'not '}a Krishnamurthy Number."
```

File: maths/perfect_number.py [hunk 6]
```diff
@@ -0,0 +1,34 @@
...
+    >>> perfect(28)
+    True
+    >>> perfect(29)
+    False
+
+    Start from 1 because dividing by 0 will raise ZeroDivisionError.
+    A number at most can be divisible by the half of the number except the number
+    itself.  For example, 6 is at most can be divisible by 3 except by 6 itself.
+    """
+    return sum(i for i in range(1, ((number // 2) + 1)) if number % i == 0) == number
```
Comment (cclauss) [inline maths/perfect_number.py:28]:
```suggestion
    return sum(i for i in range(1, number // 2 + 1) if number % i == 0) == number
```

File: maths/perfect_number.py [hunk 7]
```diff
@@ -0,0 +1,34 @@
...
+
+
+if __name__ == "__main__":
+    print("Program to check whether a number is a Perfect number or not.......")
+    number = int(input("Enter number: ").strip())
+    print(f"{number} is {'' if perfect(number) else 'not '} a Perfect Number.")
```
Comment (cclauss) [inline maths/perfect_number.py:34]:
```suggestion
    print(f"{number} is {'' if perfect(number) else 'not '}a Perfect Number.")
```

**Comment by Utsav1999** [comment]
@cclauss can you please resolve the conflict?

**Comment by Utsav1999** [comment]
Why it is giving **doctest** failure at **line 21** for **factorial(5)**?

**Comment by cclauss** [comment]
Because doctests are extremely literal.

**Pull request merged**